server/src/tests/url.test.js

```diff
@@ -0,0 +1,188 @@
+
+      expect(afterDeletingRes.status).to.equal(StatusCodes.NOT_FOUND);
+    });
+  });
+
+  describe("Gets entity by shorter url", () => {
+    it("Should return a 302 status code and redirect to the original url with the shorter url", async () => {
+      const res = await request(app).get(`/url/${urls.REAL_URL.shorterUrl}`);
+      expect(res.status).to.equal(StatusCodes.MOVED_TEMPORARILY);
+      expect(res.header["location"]).to.equal(urls.REAL_URL.originalUrl);
+    });
+
+    it("Should return a 404 status code if shorter url not found", async () => {
+      const res = await request(app).get(`/url/${urls.MOCKED_URL.shorterUrl}`);
+      expect(res.status).to.equal(StatusCodes.NOT_FOUND);
+    });
+  });
+
+  describe("Gets entities by starts with a specific string", () => {
+    it("Should return a 200 status code and all the entities that starts with a specific string", async () => {
+      const res = await request(app).get(
+        `/url/startWith/${urls.REAL_URL.shorterUrl[0]}`
+      );
+
+      expect(res.status).to.equal(StatusCodes.OK);
+      expect(res.body).to.have.lengthOf(1);
+    });
+
+    it("Should return a 404 status code if shorter urls that starts with a specific string not found", async () => {
+      const res = await request(app).get("/url/startWith/!");
+      expect(res.status).to.equal(StatusCodes.NOT_FOUND);
+    });
+  });
+
+  describe("Gets entities by contains a specific string", () => {
+    it("Should return a 200 status code and all the entities that contains a specific string", async () => {
+      const res = await request(app).get(
+        `/url/contains/${urls.REAL_URL.shorterUrl[1]}`
+      );
+
+      expect(res.status).to.equal(StatusCodes.OK);
+      expect(res.body).to.have.lengthOf(1);
+    });
+
+    it("Should return a 404 status code if shorter urls that contains a specific string not found", async () => {
+      const res = await request(app).get("/url/contains/!");
+      expect(res.status).to.equal(StatusCodes.NOT_FOUND);
+    });
+  });
+
+  describe("Gets entities that don't contain a specific string", () => {
+    it("Should return a 200 status code and all the entities that don't contain a specific string", async () => {
+      const res = await request(app).get("/url/notContains/!");
+
+      expect(res.status).to.equal(StatusCodes.OK);
+      expect(res.body).to.have.lengthOf(1);
+    });
+
+    it("Should return a 404 status code if shorter urls that don't contain a specific string not found", async () => {
+      const res = await request(app).get(
+        `/url/notContains/${urls.REAL_URL.shorterUrl[1]}`
+      );
+      expect(res.status).to.equal(StatusCodes.NOT_FOUND);
+    });
+  });
+});
+
+describe("Post", () => {
+  after(async () => {
+    await urlDeleteService.deleteUrlByShorterUrl(urls.MOCKED_URL.shorterUrl);
+  });
+
+  describe("Creates a new shorter url entity", () => {
+    it("Should return a 201 status code and the new url entity", async () => {
```

Comment:
you wrote the test checks the server returns 201 status code and the new url, why didn't you check for the new url?